test/e2e/federation-configmap.go

```diff
@@ -0,0 +1,269 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	fedclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_5"
+	"k8s.io/kubernetes/federation/pkg/federation-controller/util"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/v1"
+	kubeclientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	configmapNamePrefix       = "e2e-configmap-test-"
+	FederatedConfigMapTimeout = 60 * time.Second
+)
+
+// Create/delete configmap api objects
+var _ = framework.KubeDescribe("Federation configmaps [Feature:Federation]", func() {
+	var clusters map[string]*cluster // All clusters, keyed by cluster name
+
+	f := framework.NewDefaultFederatedFramework("federated-configmap")
+
+	Describe("ConfigMap objects", func() {
+
+		BeforeEach(func() {
+			framework.SkipUnlessFederated(f.ClientSet)
+			clusters = map[string]*cluster{}
+			registerClusters(clusters, UserAgentName, "", f)
+		})
+
+		AfterEach(func() {
+			framework.SkipUnlessFederated(f.ClientSet)
+			// Delete all configmaps.
+			nsName := f.FederationNamespace.Name
+			deleteAllConfigMapsOrFail(f.FederationClientset_1_5, nsName)
+			unregisterClusters(clusters, f)
+
+		})
+
+		It("should be created and deleted successfully", func() {
+			framework.SkipUnlessFederated(f.ClientSet)
+			nsName := f.FederationNamespace.Name
+			configmap := createConfigMapOrFail(f.FederationClientset_1_5, nsName)
+			// wait for configmap shards being created
+			waitForConfigMapShardsOrFail(nsName, configmap, clusters)
+			configmap = updateConfigMapOrFail(f.FederationClientset_1_5, nsName, configmap.Name)
+			waitForConfigMapShardsUpdatedOrFail(nsName, configmap, clusters)
+		})
+
+		It("should be deleted from underlying clusters when OrphanDependents is false", func() {
```

Comment:
I dont think we have added cascading deletion support to configmaps yet so this wont pass.